Api/RequestHandler.php

```diff
@@ -168,13 +172,13 @@ protected function buildPaymentParams( $payment )
     {
         $cc = [];
 
-        $cc['cardType']       = $this->formatCcType( $payment->getCcType() );
-        $cc['cardNumber']     = $payment->getCcNumber();
-        $cc['expirationDate'] = $this->buildCcExpDate($payment);
-        $cc['cardCode']       = $payment->getCcCid();
+        $cc['cardType']       = $this->formatCcType( $payment->getCcType() ) ?? self::DEFAULT_CARD_TYPE;
+        $cc['cardNumber']     = $payment->getCcNumber() ?? NULL;
+        $cc['expirationDate'] = $this->buildCcExpDate($payment) ?? NULL;
+        $cc['cardCode']       = $payment->getCcCid() ?? NULL;
 
         if ($last4 = $this->decryptLast4($payment)) {
-            $cc['last4'] = $last4;
+            $cc['last4'] = $last4 ?? self::DEFAULT_LAST_4;
```

Comment:
If memory serves me right `self` acts like `$this->` but for static components of the class, so we might have to make `DEFAULT_LAST_4` static in order to for this to work as expected. This would also give us some memory gains for instances of this class if this is not being used as a singleton.